components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -696,6 +700,34 @@ private String getCaptchaParams(String tenantDomain) {
                                 log.debug("Enforcing recaptcha for resend confirmation is not enabled.");
                             }
                         }
+                    } else if (captchaEnabledConfigName.equals(connectorConfig.getName())) {
+                        if (Boolean.parseBoolean(connectorConfig.getValue())) {
+                            Property[] maxFailedConfig = BasicAuthenticatorDataHolder.getInstance()
+                                    .getIdentityGovernanceService()
+                                    .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName}, tenantDomain);
+                            Property maxFailedProperty = maxFailedConfig[0];
+                            try {
+                                if (Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts) {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Number of failed attempts is higher than max failed " +
+                                                "attempts for reCaptcha. Recaptcha will be enforced");
+                                    }
+                                    captchaParams += BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
+                                } else {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Number of failed attempts is less than max failed " +
+                                                "attempts for reCaptcha. Recaptcha will not be enforced");
+                                    }
+                                }
+                            } catch (NumberFormatException e) {
+                                log.error(String.format("Invalid Max failed attempts for reCaptcha: %s",
+                                        maxFailedProperty.getValue()), e);
```

Comment:
What is the behavior if this error happens? Do we set a default value or end the flow?

Reply:
Ideally maxAttepmtBeforeSuccess claim should be an integer. But from UI, we can set non integer values as well. Hence if a non-integer charactor is set,  this error would occur. If that is the case, then we continue login procedure without enforcing the recaptcha.

Reply:
In that case don't we need to say that we are skipping the captcha flow due to this? The current error message says that it's not an int.

Reply:
fixed https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/118